wasm/datatypes/bit_size.py

```diff
@@ -1,8 +1,68 @@
 import enum
+from typing import (
+    Type,
+    Union,
+)
+
+import numpy
+
+TWrapType = Union[
+    Type[numpy.uint8],
+    Type[numpy.uint16],
+    Type[numpy.uint32],
+    Type[numpy.uint64],
+]
+TUnpack = Union[
+    numpy.int8,
+    numpy.int16,
+    numpy.int32,
+    numpy.int64,
+    numpy.uint8,
+    numpy.uint16,
+    numpy.uint32,
+    numpy.uint64,
+]
 
 
 class BitSize(enum.Enum):
-    b8 = 8
-    b16 = 16
-    b32 = 32
-    b64 = 64
+    b8 = numpy.uint8(8)
+    b16 = numpy.uint8(16)
+    b32 = numpy.uint8(32)
+    b64 = numpy.uint8(64)
+
+    def unpack_int_bytes(self, raw_bytes: bytes, signed: bool) -> TUnpack:
+        if self is self.b8:
+            if signed:
+                return numpy.frombuffer(raw_bytes, numpy.int8)[0]
```

Comment:
Might want to be careful about this since it's architecture dependent (big endian vs. little endian).